Orange/widgets/utils/itemmodels.py

```diff
@@ -849,10 +865,17 @@ def columnSortKeyData(self, column, role):
         coldesc = self.columns[column]
         if isinstance(coldesc, TableModel.Column) \
                 and role == TableModel.ValueRole:
-            col_view, _ = self.source.get_column_view(coldesc.var)
+            var = coldesc.var
+            col_view, _ = self.source.get_column_view(var)
             col_data = numpy.asarray(col_view)
             if self.__sortInd is not None:
                 col_data = col_data[self.__sortInd]
+            value_keys = self.__sortValueKeys[column]
+            if value_keys is not None:
+                if col_data is col_view:
+                    col_data = numpy.array(col_data)
+                col_data[numpy.isnan(col_data)] = len(var.values)
+                col_data = value_keys[col_data.astype(numpy.int)]
```

Comment:
This is `DiscreteVariable`, so `col_data` contains integer indices (stored as floats). It would suffice to change `int(x)` to `float(x)` in function `_create_sort_keys`. But I don't see a sensible use case.